sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -1374,12 +1374,23 @@ fn fn_signature_to_trait_fn(
     fn_signature: FnSignature,
     attributes: AttributesMap,
 ) -> Result<TraitFn, ErrorEmitted> {
-    let return_type_span = match &fn_signature.return_type_opt {
-        Some((_right_arrow_token, ty)) => ty.span(),
-        None => fn_signature.span(),
+    let return_type = match &fn_signature.return_type_opt {
+        Some((_right_arrow, ty)) => ty_to_type_argument(context, handler, engines, ty.clone())?,
+        None => {
+            let type_id = engines.te().insert(engines, TypeInfo::Tuple(Vec::new()));
+            TypeArgument {
+                type_id,
+                initial_type_id: type_id,
+                // TODO: Fix as part of https://github.com/FuelLabs/sway/issues/3635
```

Comment:
@eureka-cpu FYI, since this relates to https://github.com/FuelLabs/sway/issues/3635